docs/design-proposal.md

```diff
@@ -0,0 +1,190 @@
+# Design Proposal: Isolated Job Runner
```

Comment:
Please add the godoc for the library and the proto file in the design document, as text.

Reply:
Sure.

Reply:
Appended godoc markdown to end of design-document.md. The toc and index links work, but the tool I used to convert the godocs to markdown creates bad links within the type and function descriptions.